starport-blog/03-http-list-posts.md

```diff
@@ -0,0 +1,85 @@
+# List posts with HTTP
+
+At some point you may want to create a front-end to your application and interact with it by sending HTTP requests. In part 2 we've seen that sending requests to `/abci_query` is one way of getting data from your app. This works well for testing and certain specific usecases, but for a production application you may want to create a clean API for yourself and other users. This API should probably have short endpoints (without implementation details, like `abci`) and respond differently to HTTP verbs like GET and POST. You may also want to have some requst processing logic that handles basic checks and data transformation. Luckily, Cosmos SDK provides just that.
+
+Let's create a server that handles `GET` requests to `/blog/posts` endpoint by returning a list of posts.
+
+## `x/blog/client/rest/rest.go`
+
+```go
+import (
+  // Existing imports...
+  "fmt"
+	"net/http"
+  "github.com/cosmos/cosmos-sdk/types/rest"
+)
+```
+
+In this file you should see `func RegisterRoutes` used for registering HTTP endpoints and handler functions.
+
+Add the following line to register our first route:
+
+```go
+	r.HandleFunc("/blog/posts", listPostsHandler(cliCtx, "blog")).Methods("GET")
```

Comment:
Why is `	r.HandleFunc("/faucet", faucetHandler(cliCtx)).Methods("POST")` already included in this file?

Reply:
The template used to have a faucet, this should be removed.

https://github.com/tendermint/starport/issues/55